File: Data Structures & Algorithms/string-encode-and-decode/submission-1.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string encoded = "";
        for(string& word: strs){
            encoded += to_string(word.length()) +"#" +word;
        }

        return encoded;
    }

    vector<string> decode(string s) {
        vector<string> res;
        int itr = 0;

        while(itr<s.length()){
            int len = 0;
            while(s[itr] != '#'){
                len = len*10 + (s[itr] - '0');
                itr++;
            }
            res.push_back(s.substr(itr+1, len));
            itr += len+1;
        }

        return res;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [-1] * (n+1)
        def dfs(itr:int) -> int:
            if itr == n:
                return 1
            if itr > n:
                return 0
            if dp[itr] != -1:
                return dp[itr]
            dp[itr] = dfs(itr+1) + dfs(itr+2)
            return dp[itr]
        return dfs(0)